### ISSUE TITLE
ETQ Usager je veux pouvoir inviter quelqu’un à coconstruire mon brouillon

### ISSUE THREAD

**Issue by fredZen** [betagouv/tps#1737]
Lié à #1677

**Comment by emsnytech** [comment]
Come échangé hier, en effet, il faudra prévilégier le soin de proposer la modification des dossier en broullion ( donc non visible des agents) . Dès qu’un dossier est déposé, le sujet devient plus délicat, sauf à créer des copie des champs modifiés avec les modifications proposées, et redonner l’information au propriétaire du dossier qu’il doit valider ou pas les modifications...Cela va finir par devenir un work flow... a fuir, a moins qu’il y ait d’autres propositions

**Comment by LeSim** [comment]
je reprends après une discussion avec WorkInFrance et @emsnytech 

- un usager peut inviter une personne a co construire son dossier en brouillon
- un invité ne peut pas soumettre le dossier
- un invité peut modifier **uniquement** les dossiers en brouillon

Atterrissage souhaité  fin mai

**Comment by CamilleGarrigue** [comment]
@ ANACT
@ Agence pour la biodiversité
@ IRESP

**Comment by kemenaran** [comment]
Discuté aujourd'hui : ça semble important dans le cadre de la refonte des détails d'un dossier pour l'usager.

La question de base était : faut-il afficher un onglet "Invités" dans la nouvelle interface usager ? Si oui, ça veut dire que comme aujourd'hui on ne peut pas inviter de personne avant d'avoir soumis le formulaire, ce qui est dommage.

## Solution envisagée

- Rajouter un bandeau avec deux boutons en haut de chaque formulaire : "Notice d'aide" et "Inviter une personne à co-remplir le dossier".
- Cliquer sur le bouton "Notice d'aide" affiche la notice de la démarche.
- Cliquer sur le bouton "Inviter une personne à co-remplir le dossier" sauvegarde le brouillon, et affiche une page "Inviter une personne" avec un texte explicatif et un champ "Email".

## Questions discutées

- Pourquoi ne pas avoir directement un ajout inline dans le formulaire lui-même ? _@kemenaran et @LeSim ont peur que ça fasse trop de texte et de distraction visuelle en haut du formulaire._

**Comment by gregoirenovel** [comment]
> faut-il afficher un onglet "Invités" dans la nouvelle interface usager ? Si oui, ça veut dire que comme aujourd'hui on ne peut pas inviter de personne avant d'avoir soumis le formulaire

J'ai du mal à voir le lien entre cet onglet et la possibilité d'inviter quelqu'un (il me semble que l'invitation n'est pas sensée passer) par cet onglet.

On fait une autre issue pour la notice d'aide ?

**Comment by kemenaran** [comment]
>> faut-il afficher un onglet "Invités" dans la nouvelle interface usager ? Si oui, ça veut dire que comme aujourd'hui on ne peut pas inviter de personne avant d'avoir soumis le formulaire
> J'ai du mal à voir le lien entre cet onglet et la possibilité d'inviter quelqu'un (il me semble que l'invitation n'est pas sensée passer) par cet onglet.

Oui, y'a pas de lien direct ; c'est juste que c'est là d'où sont partis les raisonnements sur « Est-ce qu'on le fait avant la nouvelle interface des détails d'un dossier ou après ».

> On fait une autre issue pour la notice d'aide ?

Oah, moi ça me va de faire les deux d'un coup ("c'est juste un lien"). Mais si tu veux faire une issue à part pour que la réflexion soit plus carrée, ça me va aussi.

**Comment by gregoirenovel** [comment]
#2196

**Comment by kemenaran** [comment]
Mockup de l'interface pour ajouter un invité depuis un dossier en brouillon :

https://wireframe.cc/M0vlkg

<img width="1047" alt="capture d ecran 2018-07-10 a 15 52 59" src="https://user-images.githubusercontent.com/179923/42514343-79a3ce9c-8459-11e8-8910-c83bdcefab03.png">

https://wireframe.cc/2SAe9O

<img width="1047" alt="capture d ecran 2018-07-10 a 15 53 13" src="https://user-images.githubusercontent.com/179923/42514355-81508bd0-8459-11e8-8b3e-faa30f4f49c8.png">

**Comment by kemenaran** [comment]
NB : l'idée du bouton "Enregistrer un brouillon" en haut est pour l'instant au stade de la réflexion.

L'idée est que sur les formulaires très longs, l'usager ne sait pas qu'il peut enregistrer un brouillon partiel avant d'avoir scrollé tout en bas du formulaire. En mettant un bouton en haut, il le sait tout de suite.

Cela dit une meilleure solution serait peut-être de laisser tout le temps visible les boutons d'enregistrement en bas de l'écran, dans une zone flottante. C'est pas top pour les mobiles (ça réduit l'espace vertical), mais ça peut aider les autres.

**Comment by kemenaran** [comment]
> On peut inviter plusieurs personnes, cela veut dire que l’on peut avoir une liste. on peut faire apparaitre aussi un petit chiffre indiquant le nombre de personnes invitées.

Bonne remarque ; je n'ai pas fait apparaître l'état où il y a plusieurs invités sur ce mockup, mais évidemment ça incluera une liste, et une pastille avec le nombre d'invités.

> Je me demande s’il faut du texte. Ne serait t’il pas possible de reprendre les symboles que l’on connait sur les sites web (…) ?

Généralement, entre "du texte seulement", "une icône seulement", et "du texte et une icône", les conseils d'UX recommandent de mettre du texte _et_ une icône. Le texte seul n'est pas assez facile à repérer d'un seul coup d'œil, et l'icône seule n'est pas assez compréhensible par les utilisateurs.

Du coup je n'ai pas moyen de mettre d'icônes dans le logiciel de mockups – mais en vrai y'en aura, avec du texte :)

**Comment by kemenaran** [comment]
Question : un invité sur un brouillon peut éditer un formulaire. Est-ce qu'il peut aussi soumettre le dossier ?

J'imagine que oui, sinon ça ne sert pas à grand chose d'être invité. Vous en pensez quoi ?

**Comment by gregoirenovel** [comment]
Il peut éditer mais pas soumettre. C'est une limitation qui est déjà présente dans le système d'invité "hors-brouillon" il me semble

**Comment by kemenaran** [comment]
Oui, complètement.

Dans ce cas je vais rajouter un bouton "Soumettre" désactivé, avec une infobulle qui explique pourquoi.

**Comment by kemenaran** [comment]
@LeSim @emsnytech si je comprends bien, une fois que l'invitation se fait sur le brouillon, on veut désactiver la possibilité pour les invités de modifier le dossier après soumission (contrairement à ce qui se fait aujourd'hui), c'est bien ça ?

**Comment by kemenaran** [comment]
Discuté avec @tchak : légalement la modification d'un dossier soumis par une tierce-personne est compliquée. Pour lui il vaut mieux limiter.

Je pense que c'est moins bon pour le produit, mais qu'effectivement la question de la légitimité se tient. D'autres avis là dessus ?

**Comment by fredZen** [comment]
@CamilleGarrigue je te laisse transmettre à tes contacts @ ANACT, @ Agence pour la biodiversité et @ IRESP si ça a encore du sens.

@gregoirenovel je ne vois pas de rapport avec un `#2196`, ni sur github ni sur helpscout.

Je vais prochainement enlever le label `à communiquer` de ce ticket.

**Comment by gregoirenovel** [comment]
Le rapport avec #2196 c'est le commentaire de Pierre du dessus concernant la création de cette issue :)
> On fait une autre issue pour la notice d'aide ?
>> Oah, moi ça me va de faire les deux d'un coup ("c'est juste un lien"). Mais si tu veux faire une issue à part pour que la réflexion soit plus carrée, ça me va aussi.
